packages/client/src/editor/components/StickyNote.tsx

```diff
@@ -1,79 +1,74 @@
-    save: (content: string, id: UUID) => void;
-    delete: (id: UUID) => void;
+  id: UUID;
+  get: (id: UUID) => string;
+  save: (content: string, id: UUID) => void;
+  delete: (id: UUID) => void;
 }
 interface NoteState {
-    editing: boolean;
-    _newText: string
+  editing: boolean;
+  _newText: string;
 }
 
 const noteStyle = {
-    width: '200px',
-    height: '100px',
-    backgroundColor: '#f2d233',
-    transform: 'translate(400px,400px)'
-}
+  width: '200px',
+  height: '100px',
+  backgroundColor: '#f2d233',
+  transform: 'translate(400px,400px)'
+};
 
-class StickyNote extends React.Component<NoteProps, NoteState> {
-    constructor(props: NoteProps) {
-        super(props);
-        this.state = {
-            editing: false,
-            _newText: '',
-        }
-        this.save = this.save.bind(this);
-        this.edit = this.edit.bind(this);
-        this.delete = this.delete.bind(this);
-        this.renderForm = this.renderForm.bind(this);
-        this.renderDisplay = this.renderDisplay.bind(this);
-    }
+const StickyNote: React.FunctionComponent<NoteProps> = props => {
+  const [isEditing, setIsEditing] = useState(false);
 
-    edit() {
-        this.setState({editing: true});
-    }
+  // TODO this won't recevie update correctly when the note text
+  // changes not because of this form, but becaus of an update from server
+  const [newText, setNewText] = useState(props.get(props.id));
 
-    save(event: React.FormEvent<HTMLFormElement>) {
-        this.setState({_newText: event.currentTarget.value})
-        this.props.save(this.state._newText, this.props.id);
-        this.setState({editing: false});
-        event.preventDefault();
-    }
+  const edit = () => {
+    setIsEditing(true);
+  };
 
-    delete() {
-        this.props.delete(this.props.id);
-    }
+  const save = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsEditing(false);
+    props.save(newText, props.id);
+  };
 
-    renderForm() {
-        return(
-            <div style={noteStyle}>
-                    <form onSubmit={this.save}>
-                        <input type='text'/>
-                        <button type='submit'>Save</button>
-                    </form>
-            </div>
-        )
-    }
+  const deleteNote = () => {
+    props.delete(props.id);
+  };
 
-    renderDisplay() {
-        return( // Add styling for note div
-            <div style={noteStyle}>
-                <p>{this.props.get(this.props.id)}</p>
-                <button onClick={this.edit}><EditIcon/></button>
-                <button onClick={this.delete}><DeleteIcon/></button>          
-            </div>
-        )
-    }
-    
-    render() {
-        return this.state.editing ? this.renderForm() : this.renderDisplay();
-    }
-}
+  if (isEditing) {
+    return (
+      <div style={noteStyle}>
+        <form onSubmit={save}>
+          <input
+            type="text"
+            value={newText}
```

Comment:
Providing `value` here is necessary to have the current text in the input rather than empty string.

Generally, recommended reading: [Forms](https://reactjs.org/docs/forms.html)